owncloud/web
# feat: move access token timer to web worker

## Description
Introduces a web worker for the access token timer to run in. As soon as a token is about to expire, the web worker will inform the main thread that it needs to retrieve a new token from the server.

This ensures that the timer runs es expected even if the browser tab is being throttled or suspended. As a result, long running operation such as uploads or move actions can continue in the background.

## Related Issue




- refs https://github.com/owncloud/web/issues/10855#issuecomment-2126553354
- fixes https://github.com/owncloud/web/issues/10474 (hopefully)

## How has this been tested

- Set `IDP_ACCESS_TOKEN_EXPIRATION` to `120`
- Add a time log to the user loaded event (takes place immediately after each token renewal)
- Leave the browser tab open in background, do other things
- Check the tab after some time period (~ 15-20 min)
- The time difference between the time logs should be a consistent 110 + 2-3 seconds: 120 seconds minus 10 seconds (because we're triggering the renewal 10 seconds before expiry) plus 2-3 seconds for the token renewal to take place (this varies a bit because of server & network)

## Types of changes

- [ ] Maintenance (e.g. dependency updates or tooling)
